src/main/java/com/gard/testbed/activitySystem/Goal.java

```diff
@@ -0,0 +1,16 @@
+package com.gard.testbed.activitySystem;
+
+import java.util.function.Predicate;
+
+/**
+ * Created by va922kg on 3/15/16.
+ */
+public class Goal {
```

Comment:
I didn't use the goal yet, the idea is to have a GoalTransition that accept a Goal as constructor parameter.
Not every transition is goal related, some of them are failure